### PR TITLE
[NV-3328] fix(web,api,webhook): add missing webhook field in the integration sidebar

### PR DESCRIPTION
**Pull request by LetItRock**
### What change does this PR introduce?

Add the missing webhook URL field in the SMS/Email integration sidebar.

### Other information (Screenshots)

Screenshot 2023-10-06 at 17 05 47

https://github.com/novuhq/novu/assets/2607232/b1216ce8-8a48-41a6-a887-880b34426713

### CODE REVIEW

File: apps/api/src/app/integrations/e2e/get-webhook-support-status.e2e.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+  ChannelTypeEnum,
+  ChatProviderIdEnum,
+  EmailProviderIdEnum,
+  InAppProviderIdEnum,
+  PushProviderIdEnum,
+  SmsProviderIdEnum,
+} from '@novu/shared';
+import { IntegrationEntity, IntegrationRepository } from '@novu/dal';
+
+describe('Get Webhook Support Status - /webhook/provider/:providerOrIntegrationId/status (GET)', function () {
```
Comment (LetItRock) [inline apps/api/src/app/integrations/e2e/get-webhook-support-status.e2e.ts:13]: missing tests

File: apps/api/src/app/integrations/usecases/get-webhook-support-status/get-webhook-support-status.usecase.ts [hunk 2]
```diff
@@ -15,14 +16,17 @@ export class GetWebhookSupportStatus {
...
-
     const integration = await this.getIntegration(command);
     if (!integration) {
-      throw new NotFoundException(`Integration for ${providerId} was not found`);
+      throw new NotFoundException(`Integration for ${command.providerOrIntegrationId} was not found`);
+    }
+
+    const hasNoCredentials = !integration.credentials || Object.keys(integration.credentials).length === 0;
+    if (hasNoCredentials) {
+      throw new ApiException(`Integration ${integration._id} doesn't have credentials set up`);
```
Comment (LetItRock) [inline apps/api/src/app/integrations/usecases/get-webhook-support-status/get-webhook-support-status.usecase.ts:26]: the credentials are required to build the provider

File: apps/web/src/api/hooks/useWebhookSupportStatus.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import { useQuery } from '@tanstack/react-query';
+import { ChannelTypeEnum } from '@novu/shared';
+
+import { getWebhookSupportStatus } from '../integration';
+import { IS_DOCKER_HOSTED, WEBHOOK_URL } from '../../config';
+import { useAuthController, useEnvController } from '../../hooks';
+
+export const useWebhookSupportStatus = ({
```
Comment (LetItRock) [inline apps/web/src/api/hooks/useWebhookSupportStatus.ts:8]: the hook that verifies if the integration supports parsing events with the webhook

File: apps/web/src/pages/integrations/IntegrationsListModal.tsx [hunk 4]
```diff
@@ -187,6 +187,7 @@ export function IntegrationsListModal({
         channel={provider?.channel}
       />
       <UpdateProviderSidebar
+        key={integrationIdToEdit}
```
Comment (LetItRock) [inline apps/web/src/pages/integrations/IntegrationsListModal.tsx:190]: Fix the issue when the Update Integration Sidebar is not updated when the different integration is selected.

File: apps/web/src/pages/integrations/components/multi-provider/UpdateProviderSidebar.tsx [hunk 5]
```diff
@@ -357,6 +369,20 @@ export function UpdateProviderSidebar({
...
+                value={webhookUrl}
+                readOnly
+                rightSection={
+                  <CopyWrapper onClick={() => webhookUrlClipboard.copy(webhookUrl)}>
+                    {webhookUrlClipboard.copied ? <Check /> : <Copy />}
+                  </CopyWrapper>
+                }
+              />
+            </InputWrapper>
+          )}
```
Comment (LetItRock) [inline apps/web/src/pages/integrations/components/multi-provider/UpdateProviderSidebar.tsx:385]: missing Webhook URL field

File: apps/webhook/src/webhooks/usecases/webhook/webhook.usecase.ts [hunk 6]
```diff
@@ -26,44 +39,51 @@ export class Webhook {
   ) {}
 
   async execute(command: WebhookCommand): Promise<IWebhookResult[]> {
-    const providerId = command.providerId;
+    const providerOrIntegrationId = command.providerOrIntegrationId;
```
Comment (LetItRock) [inline apps/webhook/src/webhooks/usecases/webhook/webhook.usecase.ts:42]: the webhook usecase that parses events

**Comment by scopsy** [comment]
@LetItRock could we try to merge this? Been open for a while now

**Review by BiswaViraj — APPROVED** [review]
🚀

**Comment by BiswaViraj** [comment]
@LetItRock nice work 💪🏼 , can you please check the failing tests

**Pull request merged**